docs/contributing/database-migrations/edd.mdx

```diff
@@ -173,49 +222,82 @@ END
+---
 
-### Developer
+![Bitwarden EDD State Machine](./edd_state_machine.jpg) \[Open Image in a new tab for better
+viewing\]
 
-The development flow is described in [Migrations](./).
+---
 
-### Devops
+### Cloud Environments
 
-#### On `rc` cut
+Since we treat both schema migrations and data migrations as just migrations, the only issues that
+we are solving for is orchestrating the runtime constraints on the migration. Eventually, all
+migrations will end up in `DbScripts`. However, to control the timing of running _Transition_ and
+associated _Finalization_ migrations, we need to keep them outside of `DbScripts` until the correct
+timing.
 
-Create a PR moving the future scripts.
+In our environments with always-on applications, _Transition_ scripts must be run after the new code
+has been rolled out. To execute a full deploy, we run all new migrations in `DbScripts`, roll out
+the new code, and then run all _Transition_ migrations in the `DbScripts_transition` directory as
+soon as all of the new code services are online. In the case of a critical failure after the new
+code is rolled out, we will conduct a Rollback (see Rollbacks below). _Finalization_ migrations will
+not be run until the start of the next deploy when they are moved into `DbScripts`.
 
-- `DbScripts_future` to `DbScripts`, prefix the script with the current date, but retain the
-  existing date.
-- `dbo_future` to `dbo`.
-  <bitwarden>
-    <li>
-      Create a ticket in Jira with a `Due Date` of the release date to ensure future migrations are
-      merged in and ready to be executed. Set the ticket that created the future migration as a
-      blocker.
-    </li>
-  </bitwarden>
+After this deploy, to prep for the next release, all migrations in `DbScripts_transition` are moved
+to `DbScripts` and then all migrations in `DbScripts_finalization` are moved to `DbScripts`,
+conserving their execution order for a clean install. For our current branching strategy, PRs will
+be open against `master` when `rc` is cut to prep for this release. This PR automation will also
+handle renaming the migration file and updating any reference of `[dbo_future]` to `[dbo]`.
 
-#### After server release
+The next deploy will pick up the newly added migrations in `DbScripts` and set the previously
+repeatable _Transition_ migrations to no longer be repeatable, execute the _Finalization_
+migrations, and then execute any new migrations associated with the code changes that are about to
+go out.
 
-1. Run whatever data migration scripts might be needed. (This might need to be batched and executed
-   until all the data has been migrated)
-2. After having the server run for a while execute the future migration script to clean up the
-   database.
+The the state of migrations in the different directories at any one time is is saved and versioned
+in our Migrator Utility which supports the EDD phased migration process in both types of
+environments.
+
+### Standard Self-host Environments
```

Comment:
💭 Per the other comment. This could be "offline", and I would drop "standard".

Reply:
Done and done.